contracts/BufferLib.sol

```diff
@@ -52,6 +52,7 @@ library BufferLib {
     uint64 newCursor = _offset;
     // Deal with relative offsets
     if (_relative == true) {
+      require(newCursor + _buffer.cursor > newCursor, "The sum newCursor caused an overflow");
```

Comment:
Maybe a more detailed error message (i.e. stating the function where it occurred)
```suggestion
      require(newCursor + _buffer.cursor > newCursor, "Integer overflow when seeking");
```